lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -56,6 +114,500 @@ static Value adjustIntegerWidth(OpBuilder &builder, Value value,
+
+      auto b = ifOp.getThenBodyBuilder();
+      trueValue = rewriter.getRemappedValue(trueValue);
+      falseValue = rewriter.getRemappedValue(falseValue);
+
+      if (allConds.size() > 1) {
+        cond = b.create<comb::AndOp>(ifOp.getLoc(), allConds, false);
+        allConds.pop_back();
+      }
+      auto muxOp =
+          b.create<comb::MuxOp>(ifOp.getLoc(), cond, trueValue, falseValue);
+      muxMap[lhs] = muxOp;
+    }
+
+    // Handle '? :'.
+    if (isa<scf::YieldOp>(thenOp) && thenOp.getNumOperands()) {
+      trueValue = thenOp.getOperand(0);
+      rewriter.inlineBlockBefore(&ifOp.getThenRegion().front(), ifOp);
+      rewriter.inlineBlockBefore(&ifOp.getElseRegion().front(), ifOp);
+
+      trueValue = rewriter.getRemappedValue(trueValue);
+      falseValue = rewriter.getRemappedValue(falseValue);
+
+      if (allConds.size() > 1) {
+        cond = rewriter.create<comb::AndOp>(ifOp.getLoc(), allConds, false);
+        allConds.pop_back();
+      }
+      rewriter.replaceOpWithNewOp<comb::MuxOp>(ifOp, cond, trueValue,
+                                               falseValue);
+      return success();
+    }
+  }
+  rewriter.inlineBlockBefore(&ifOp.getThenRegion().front(), ifOp);
+  if (!ifOp.getElseRegion().empty()) {
+    rewriter.inlineBlockBefore(&ifOp.getElseRegion().front(), ifOp);
+  }
+  rewriter.eraseOp(ifOp);
+  return success();
+}
+
+/// Handle "always_comb".
+static LogicalResult lowerAlwaysComb(ProcedureOp procedureOp,
+                                     ConversionPatternRewriter &rewriter) {
+  DenseMap<Operation *, Value> muxMap;
+  for (auto &nestOp : procedureOp.getBodyRegion().getOps()) {
+    SmallVector<Value> allConds;
+    if (isa<scf::IfOp>(nestOp))
+      if (failed(lowerSCFIfOp(dyn_cast<scf::IfOp>(nestOp), rewriter, muxMap,
+                              allConds)))
+        return failure();
+  }
+
+  // Update the values of the variables/nets.
+  for (auto muxOp : muxMap) {
+    auto from = rewriter.getRemappedValue(muxOp.first->getResult(0));
+    if (!from)
+      return failure();
+
+    rewriter.replaceUsesWithIf(from.getDefiningOp()->getOperand(0),
+                               muxOp.second, [&](OpOperand &use) {
+                                 Operation *user = use.getOwner();
+                                 return user == from.getDefiningOp();
+                               });
+  }
+
+  rewriter.inlineBlockBefore(procedureOp.getBody(), procedureOp);
+  return success();
+}
+
+/// Lower `always` with an explicit sensitivity list and clock edges to `seq`,
+/// otherwise to `comb`.
+static LogicalResult lowerAlways(ProcedureOp procedureOp,
+                                 ConversionPatternRewriter &rewriter) {
+  // The default is a synchronization.
+  bool isAsync = false;
+
+  // The default is the combinational logic unit.
+  bool isCombination = true;
+
+  // Assume the reset value is at the then region.
+  bool atThenRegion = true;
```

Comment:
Unused. But we will implement the code to judge which region is the reset value as quickly as possible. Example a simple DFF.
```
input rst;
always @(posedge clk)   //synchronic
//  two different situations.
if(!rst) // at else region   or   if(rst) // at then region 
```